### ISSUE TITLE
Some lifecycle date-related facets ("last accessioned", etc.) disappear mysteriously from search and/or facet result pages

### ISSUE THREAD

**Issue by andrewjbtw** [sul-dlss/argo#4165]
**Describe the bug**
The set of date-related facets in Argo includes:

- registered
- last accessioned
- earliest accessioned
- last published
- last submitted
- last ingested
- last modified
- last opened

Almost every accessioned object should have metadata corresponding to each of these facets. But sometimes when running a search only a subset of these facets appears. I'd like to understand why the other facets don't appear.

Example: [Search for "goethe" as a keyword](https://argo.stanford.edu/catalog?q=goethe&search_field=text)

When I run that search, I only see two of the date-related facets:

- earliest accessioned
- last opened

Those items should have metadata that corresponds to other dates in the facet list.

**User Impact**
Users cannot rely on the date facets to be consistent. For example, for QC or data analysis purposes, I may want to find everything that was recently accessioned. I can't do that if the "last accessioned" facet doesn't capture the last accessioned dates of all of the druids. Similarly, it is not possible to run a search and then filter further by facet if some of those facets do not appear.

**Expected behavior**
Facets that correspond to dates/times in the object registration and accessioning lifecycle should include all items that have gone through those lifecycles. For example, every accessioned object should have

- earliest accessioned date (date when version 1 was accessioned)
- last accessioned date (date when the most recent version was accessioned)

We do have some gaps in our data such as items for which no registered date was recorded, but we should have more data than the facets currently include.

**Additional context**
We may remove some of these facets in the future. As things stand, I make use of whichever facet happens to be available if the facet I actually want doesn't appear. If we could reliably make a consistent subset of the facets appear, we could potentially remove other facets.

**Comment by ndushay** [comment]
Thank you, Justin - very helpful.

So if Andrew wants some of these date facets to at least provide the search box for exploration even if there are no results, then he will ticket.

For example:  

searching on "goethe" there is an earliest accessioned facet showing with 139 items that were accessioned for the first time in the last year.

Using the range selection for the earliest accessioned facet, we chose 2015 to 2021.   We got results.  But the earliest accessioned facet disappeared.

https://argo.stanford.edu/catalog?q=goethe&search_field=text&f%5Baccessioned_earliest_dttsi%5D%5B%5D=%5B2015-01-01T00%3A00%3A00.000Z+TO+2021-01-01T23%3A59%3A59.000Z%5D&commit=submit

So we designed Argo facets this way (or blacklight did).

Let's leave this ticket open and assign it to Andrew.
